fe/fe-core/src/test/java/org/apache/doris/nereids/rules/exploration/join/JoinLAsscomTest.java

```diff
@@ -0,0 +1,152 @@
+import mockit.Mocked;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Optional;
+
+public class JoinLAsscomTest {
+    public Pair<LogicalJoin, LogicalJoin> testJoinLAsscom(PlannerContext plannerContext,
+            Expression bottomJoinOnCondition, Expression topJoinOnCondition) {
+        /*
+         *      topJoin                newTopJoin
+         *      /     \                 /     \
+         * bottomJoin  C   -->  newBottomJoin  B
+         *  /    \                  /    \
+         * A      B                A      C
+         */
+        Table t1 = new Table(0L, "t1", Table.TableType.OLAP,
+                ImmutableList.of(new Column("id", Type.INT, true, AggregateType.NONE, "0", ""),
+                        new Column("name", Type.STRING, true, AggregateType.NONE, "0", "")));
+        LogicalOlapScan scan1 = new LogicalOlapScan(t1, ImmutableList.of());
+
+        Table t2 = new Table(0L, "t2", Table.TableType.OLAP,
+                ImmutableList.of(new Column("id", Type.INT, true, AggregateType.NONE, "0", ""),
+                        new Column("name", Type.STRING, true, AggregateType.NONE, "0", "")));
+        LogicalOlapScan scan2 = new LogicalOlapScan(t2, ImmutableList.of());
+
+        Table t3 = new Table(0L, "t3", Table.TableType.OLAP,
+                ImmutableList.of(new Column("id", Type.INT, true, AggregateType.NONE, "0", ""),
+                        new Column("name", Type.STRING, true, AggregateType.NONE, "0", "")));
+        LogicalOlapScan scan3 = new LogicalOlapScan(t3, ImmutableList.of());
+
+        LogicalJoin<LogicalOlapScan, LogicalOlapScan> bottomJoin = new LogicalJoin<>(JoinType.INNER_JOIN,
+                Optional.of(bottomJoinOnCondition), scan1, scan2);
+        LogicalJoin<LogicalJoin<LogicalOlapScan, LogicalOlapScan>, LogicalOlapScan> topJoin = new LogicalJoin<>(
+                JoinType.INNER_JOIN, Optional.of(topJoinOnCondition), bottomJoin, scan3);
+
+        Rule rule = new JoinLAsscom().build();
+        List<Plan> transform = rule.transform(topJoin, plannerContext);
+        Assert.assertEquals(1, transform.size());
+        Assert.assertTrue(transform.get(0) instanceof LogicalJoin);
+        LogicalJoin newTopJoin = (LogicalJoin) transform.get(0);
+        return new Pair<>(topJoin, newTopJoin);
+    }
+
+    @Test
+    public void testStarJoinLAsscom(@Mocked PlannerContext plannerContext) {
+        /*
+         * Star-Join
+         * t1 -- t2
+         * |
+         * t3
+         * <p>
+         *     t1.id=t3.id               t1.id=t2.id
+         *       topJoin                  newTopJoin
+         *       /     \                   /     \
+         * t1.id=t2.id  t3          t1.id=t3.id   t2
+         * bottomJoin       -->    newBottomJoin
+         *   /    \                   /    \
+         * t1      t2               t1      t3
+         */
+        Expression bottomJoinOnCondition = new EqualTo(
```

Comment:
u should set ExprId manually to ensure same SlotReference in child output and join Condition has the same ID. In real world this is always true.